### PR TITLE
Rename Devtool to Hook

### PR DESCRIPTION
**Pull request by keyanzhang**
Now `devtool` stands for our [chrome extension](https://github.com/facebook/react-devtools/) and our internal stuff are called `hook`s.

Module name changes:
```
ReactComponentTreeDevtool -> ReactComponentTreeHook
ReactDOMNullInputValuePropDevtool -> ReactDOMNullInputValuePropHook
ReactDOMUnknownPropertyDevtool -> ReactDOMUnknownPropertyHook
ReactChildrenMutationWarningDevtool -> ReactChildrenMutationWarningHook
ReactHostOperationHistoryDevtool -> ReactHostOperationHistoryHook
ReactInvalidSetStateWarningDevTool -> ReactInvalidSetStateWarningHook
```

Internal API changes:
```
ReactDebugTool.addDevtool -> ReactDOMDebugTool.addHook
ReactDebugTool.removeDevtool -> ReactDOMDebugTool.removeHook
ReactDOMDebugTool.addDevtool -> ReactDOMDebugTool.addHook
ReactDOMDebugTool.removeDevtool -> ReactDOMDebugTool.removeHook
```

Note: https://github.com/facebook/react/pull/7381/commits/bba0d992d8f4ecf9cf6677817a1218e7f48a8a77 is a temporary compatibility fix for React Native/www.

### CODE REVIEW

File: src/isomorphic/hooks/ReactComponentTreeDevtool.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule ReactComponentTreeDevtool
+ */
+'use strict';
+
+// TODO remove this proxy when RN/www gets updated
```
Comment (gaearon) [inline src/isomorphic/hooks/ReactComponentTreeDevtool.js]: If we are trying to keep it backward compatible, we also have to allow old method names (addDevtool, removeDevtool).

Comment (keyanzhang) [inline src/isomorphic/hooks/ReactComponentTreeDevtool.js]: yeah, I added backward compatibility support for `ReactDebugTool` in https://github.com/facebook/react/pull/7381/commits/bba0d992d8f4ecf9cf6677817a1218e7f48a8a77. Not sure if we should also do it for `ReactDOMDebugTool` tho :)

**Comment by gaearon** [comment]
LGTM

**Comment by zpao** [comment]
Going to just say this is patch-compatible since we have the compat layer in there. It's not a real public API so not a big deal as long as we aren't breaking RN.

**Pull request merged**